src/Features/VisualBasic/Portable/ValidateFormatString/VisualBasicValidateFormatStringDiagnosticAnalyzer.vb

```diff
@@ -0,0 +1,53 @@
+﻿'Copyright(c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt In the project root For license information.
+
+Imports Microsoft.CodeAnalysis.Diagnostics
+Imports Microsoft.CodeAnalysis.LanguageServices
+Imports Microsoft.CodeAnalysis.ValidateFormatString
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.ValidateFormatString
+
+    <DiagnosticAnalyzer(LanguageNames.VisualBasic)>
+    Friend Class VisualBasicValidateFormatStringDiagnosticAnalyzer
+        Inherits AbstractValidateFormatStringDiagnosticAnalyzer(Of SyntaxKind)
+
+        Protected Overrides Function GetSyntaxFactsService() As ISyntaxFactsService
+            Return VisualBasicSyntaxFactsService.Instance
+        End Function
+
+        Protected Overrides Function GetInvocationExpressionSyntaxKind() As SyntaxKind
+            Return SyntaxKind.InvocationExpression
+        End Function
+
+        Protected Overrides Function GetArgumentExpressionType(
+                semanticModel As SemanticModel,
+                argsArgument As SyntaxNode) As ITypeSymbol
+
+            Dim argumentSyntax = DirectCast(argsArgument, ArgumentSyntax)
+            Return semanticModel.GetTypeInfo(argumentSyntax.GetArgumentExpression()).Type
+        End Function
+
+        Protected Overrides Function GetMatchingNamedArgument(
+                arguments As SeparatedSyntaxList(Of SyntaxNode),
+                searchArgumentName As String) As SyntaxNode
+
+            For Each argument In arguments.Cast(Of ArgumentSyntax)
+                Dim simpleArgumentSyntax = DirectCast(argument, SimpleArgumentSyntax)
```

Comment:
@CyrusNajmabadi , I changed this to TryCast in the next PR, but I have a couple of questions for you...

If it's not SimpleArgumentSyntax, then it would have to be either RangeArgumentSyntax or OmittedArgumentSyntax (if I'm missing others let me know).  Is it possible to pass a RangeArgumentSyntax to String.Format?  And when passing OmittedArgumentSyntax, I called GetSymbolInfo on the expression earlier in the code, which returns null and exits the analyzer .  The specific test is: String.Format(,)    Is there another test case I should be running?